pkg/sql/opt/optgen/cmd/langgen/BUILD.bazel

```diff
@@ -1,5 +1,31 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
 
+# The Optgen language compiler uses itself to generate its AST expressions.
+# This is a form of compiler bootstrapping[1], and needs some manual overrides
+# to fit into how Bazel wants things to be. We define a second go_library
+# target[2], that sources pre-generated og.go files that are already checked
+# into the source tree. These files are used to compile langgen, which is then
+# used to generate newer revisions of those same generated files. All
+# dependants of [3] will need to depend on the og.go files generated in the
+# Bazel sandbox, which is achieved by [4]. Conversely, to instruct
+# gazelle/Bazel to resolve langgen's import of [3] appropriately, we add the
+# resolve directive below.
+#
+# TODO(irfansharif): We'll eventually want to be able to implant the sandbox
+# generated files back into the source tree. This will be necessary for IDEs,
+# but also for a sane workflow for the engineers working on langgen. If they
+# were only using Bazel today, langgen dependent builds would misbehave as we
+# wouldn't be updating the checked-in og.go files used to build langgen itself
+# (assuming any diffs there would result in different code/compilation
+# behaviour). See https://github.com/cockroachdb/cockroach/issues/58018.
```

Comment:
Not sure why lint checking in teamcity is taking "Behaviour" as an error if:
[Behavior is the preferred spelling in American English. Behaviour is preferred everywhere else](https://grammarist.com/spelling/behavior-behaviour/).

Reply:
Pretty annoying linter rule to have working for an american company.